src/ElboDeriv.jl

```diff
@@ -151,6 +166,52 @@ function ElboIntermediateVariables(
 end
 
 
+"""
+Add all the elbo values for an elbo_vars_array in the first element.
+After a value is added, it is cleared.
+"""
+function reduce_elbo_vars_array!{NumType <: Number}(
+    elbo_vars_array::Array{ElboIntermediateVariables{NumType}};
+    num_threads::Int=nthreads())
+
+  if Threaded
```

Comment:
@rgiordan: I suggest eliminating the test for ``Threaded``, because if the flag is turned off, ``test_elbo_values.jl`` fails as follows:
```
Testing b = 1
Testing b = 2
Testing b = 3
Testing b = 4
Testing b = 5

1
2
3
        FAILED: test_elbo_values.jl
ERROR: LoadError: LoadError: assertion failed: |elbo_vars_array[1].elbo.v - total_elbo.v| <= 7.450580596923828e-5
  elbo_vars_array[1].elbo.v = [-2.505919163510873e6]
  total_elbo.v = [-6.414734300769643e7]
  difference = 6.164142384418555e7 > 7.450580596923828e-5
 in error(::ASCIIString, ::ASCIIString, ::Vararg{Any}) at ./error.jl:22
 in test_approx_eq(::Array{Float64,1}, ::Array{Float64,1}, ::Float64, ::ASCIIString, ::ASCIIString) at ./test.jl:751
 [inlined code] from ./math.jl:180
 in test_approx_eq(::Array{Float64,1}, ::Array{Float64,1}, ::ASCIIString, ::ASCIIString) at ./test.jl:761
 in test_reduce_elbo_vars_array() at /global/u1/k/kpamnany/.julia/v0.5/Celeste/test/test_elbo_values.jl:407
 in include(::ASCIIString) at ./boot.jl:264
 in include_from_node1(::ASCIIString) at ./loading.jl:417
 [inlined code] from /global/u1/k/kpamnany/.julia/v0.5/Celeste/test/runtests.jl:40
 in anonymous at ./no file:4294967295
 in include(::ASCIIString) at ./boot.jl:264
 in include_from_node1(::UTF8String) at ./loading.jl:417
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
while loading /global/u1/k/kpamnany/.julia/v0.5/Celeste/test/test_elbo_values.jl, in expression starting on line 429
while loading /global/u1/k/kpamnany/.julia/v0.5/Celeste/test/runtests.jl, in expression starting on line 38
```